enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/ThreadedTransaction.java

```diff
@@ -102,22 +104,34 @@ public Throwable apply( S subject )
                                 if ( !startEarly )
                                 {
                                     latch.startAndWaitForAllToStart();
+                                    state = LatchState.STARTED;
                                 }
                             }
                             finally
                             {
-                                if ( !startEarly )
+                                if ( !startEarly && state == LatchState.NOT_STARTED )
                                 {
                                     latch.start();
+                                    state = LatchState.STARTED;
                                 }
                                 latch.finishAndWaitForAllToFinish();
+                                state = LatchState.FINISHED;
                             }
                             result.close();
                             tx.success();
                             return null;
                         }
                         catch (Throwable t)
                         {
+                            if ( state == LatchState.NOT_STARTED )
+                            {
+                                latch.start();
+                                state = LatchState.STARTED;
+                            }
+                            if ( state == LatchState.STARTED )
+                            {
+                                latch.finish();
+                            }
```

Comment:
You're correct in part, there's some redundant checks in this block, since it will always be in one of two states. But it's required that the latch be counted down since otherwise the tests will hang if there's an error in the outer try block.